frontend/src/hooks/queryBuilder/useSetCurrentKeyAndOperator.ts

```diff
@@ -1,27 +1,27 @@
-import { AttributeKeyOptions } from 'api/queryBuilder/getAttributesKeysValues';
+import { getTagToken } from 'container/QueryBuilder/filters/QueryBuilderSearch/utils';
 import { useMemo } from 'react';
+import { BaseAutocompleteData } from 'types/api/queryBuilder/queryAutocompleteResponse';
 import { getCountOfSpace } from 'utils/getCountOfSpace';
-import { separateSearchValue } from 'utils/separateSearchValue';
 
 type ICurrentKeyAndOperator = [string, string, string[]];
 
 export const useSetCurrentKeyAndOperator = (
 	value: string,
-	keys: AttributeKeyOptions[],
+	keys: BaseAutocompleteData[],
 ): ICurrentKeyAndOperator => {
 	const [key, operator, result] = useMemo(() => {
 		let key = '';
 		let operator = '';
 		let result: string[] = [];
 
 		if (value) {
-			const [tKey, tOperator, tResult] = separateSearchValue(value);
-			const isSuggestKey = keys?.some((el) => el.key === tKey);
+			const { tagKey, tagOperator, tagValue } = getTagToken(value);
+			const isSuggestKey = keys?.some((el) => tagKey.includes(el.key));
 
 			if (getCountOfSpace(value) >= 1 || isSuggestKey) {
-				key = tKey || '';
-				operator = tOperator || '';
-				result = tResult.filter((el) => el);
+				key = tagKey || '';
+				operator = tagOperator || '';
+				result = tagValue || [];
```

Comment:
What is happening here? What is `getCountOfSpace` or `isSuggestKey` doing?

Reply:
@srikanthccv - Removed space logic
https://github.com/techchintan/signoz/commit/ec381985ecc9812e83e0cca23990d3543f57b341